### PR TITLE
Do not sort joins in history

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Unblocks progress on https://github.com/rchain/rchain/pull/2419. Makes bonding work again

### JIRA ticket:
Related to: https://rchain.atlassian.net/browse/RCHAIN-3327

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala [hunk 1]
```diff
@@ -214,9 +223,8 @@ object HistoryRepositoryImpl {
       .encode(
         joins
           .map(
-            channels => encodeSorted(channels)
+            channels => encodeUnsorted(channels)
```
[the remark this answers is not in the document]
Comment (ArturGajowy) [reply rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala]: I think adding a comment in the commit message what bug / test failure led to the discovery of the new correct behavior would help me understand - but I'm fine if it's just in the PR

**Comment by lukasz-golebiewski** [comment]
bors retry

**Pull request merged**